### PR TITLE
Generate a list of clint errors

### PR DESCRIPTION
**Pull request by ZyX-I**
Ref neovim/neovim#3174

Hope I have not messed anything up because I cannot really test this.

### CODE REVIEW

File: ci/clint-errors.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+  ./clint.py --record-errors="$errors_file" \
+    src/nvim/**/*.c src/nvim/**/*.h 2>> "$index_file" || true
+
+  local title="Clint.py errors list"
+  local body="<pre>$(cat "$index_file")</pre>"
+  generate_report "$title" "$body" "$index_file"
+}
+
+# Helper function to download clang analyzer badge from shields.io
```
Comment (justinmk) [inline ci/clint-errors.sh]: clang => clint

**Comment by justinmk** [comment]
Looks fine to me. For files that are not currently blacklisted, if a clint error occurs, will it fail the build at `neovim/neovim`? I think it should.

**Comment by ZyX-I** [comment]
It will fail, but not if file with the error was not merged to master. After error was merged it will no longer fail up until some adjacent line is not modified.

**Comment by justinmk** [comment]
> It will fail, but not if file with the error was not merged to master.

Do you mean *was merged* to master? We need style errors to be caught in PRs, at least for the files that *already* pass clint.

**Comment by ZyX-I** [comment]
@justinmk `bot-ci` updates the list of suppressed errors from the current master. If somebody *ignored* the travis build result and merged file with the error into the master, this error will no longer fail the build in the other PRs.

If travis clint build results are not ignored then no new errors are going to leak in.

**Comment by justinmk** [comment]
:+1: Perfect

**Comment by justinmk** [comment]
Here it is: http://neovim.io/doc/reports/clint/

**Comment by ZyX-I** [comment]
@justinmk Actually I *have* messed things up: I forgot that `src/nvim/**/*.c src/nvim/**/*.h` is not going to work without `shopt -s globstar`.

**Comment by ZyX-I** [comment]
@justinmk Actually errors file should contain more then eleven thousands of errors.

**Comment by justinmk** [comment]
I thought recent bash versions supported that by default, but ok.

**Pull request merged**